### PR TITLE
Call Start Change

### PR DESCRIPTION
**Pull request by SpadesNeil**
This PR changes the order of the call start. Instead of checking if we have the correct caller, we first check if we have the incorrect caller.

My argument for this is people have thousands of different ways of answering the phone from the sensible to the insane. We have also lost a number of calls during into just because the speech reco was being dumb. If it's the wrong person and not a spouse, they are going to hang up anyway.

~~Meanwhile, things like wrong numbers and answering machines are already handled elsewhere in the code.~~ Actually let's handle those here too, to waste less time. I feel that the benefits heavily outweigh the problems that would happen with this change.

**tl;dr** Rather than look for "yes" or "this is him" etc, which is only a very finite set of good results, instead this uses a process of elimination.

### CODE REVIEW

**Comment by SpadesNeil** [comment]
Holding off on testing this until issues in the main branch are resolved and we can confirm our call list isn't awful.

**Comment by SpadesNeil** [comment]
I'm going to test this on my own computer tonight to make absolutely sure it's working as intended. Already I suspect there will be issues mainly with answering machines, but there's already checks for those, I'll just add in any missing ones.

**Comment by SpadesNeil** [comment]
Good news! This works exactly as it has intended. I've been live auditing it. The only reason it has no leads yet is there _are_ no valid leads yet that have come in.

*However* it has fixed exactly the problem I set out to fix. The intro of calls comes out as all manner of unexpected results, which we are hopeless to account for. Instead, this new system uses a process of elimination and calls that otherwise would have been dropped immediately by Cheryl going, "Sorry can you repeat that please?" were not dropped.

Removing "Needs Testing" tag. The proof of concept works as intended.
